src/core/batch.rkt

```diff
@@ -83,6 +85,14 @@
     (timeline-push! 'compiler size (batch-length final)))
   final)
 
+(define (batch-extract-exprs b roots)
```

Comment:
This function is the same as `batch->progs`, however for specific `roots`, not just default `batch-roots`